pkg/services/ngalert/ngalert.go

```diff
@@ -0,0 +1,188 @@
+func (ng *AlertNG) LoadAlertCondition(alertDefinitionID int64, signedInUser *models.SignedInUser, skipCache bool) (*eval.Condition, error) {
+	getAlertDefinitionByIDQuery := GetAlertDefinitionByIDQuery{ID: alertDefinitionID}
+	if err := bus.Dispatch(&getAlertDefinitionByIDQuery); err != nil {
+		return nil, err
+	}
+	alertDefinition := getAlertDefinitionByIDQuery.Result
+
+	condition := eval.Condition{RefID: alertDefinition.Condition}
+	var ds *models.DataSource
+	for _, query := range alertDefinition.Data {
+		model := make(map[string]interface{})
+		err := json.Unmarshal(query.JSON, &model)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal query model %w", err)
+		}
+
+		i := model["datasource"]
+		dsName, _ := i.(string)
+		if dsName != "__expr__" {
+			i, ok := model["datasourceId"]
+			if !ok {
+				return nil, fmt.Errorf("failed to get datasourceId from query model")
+			}
+			datasourceID, ok := i.(float64)
+			if !ok {
+				return nil, fmt.Errorf("failed to cast datasourceId to int64")
+			}
+
+			ds, err = ng.DatasourceCache.GetDatasource(int64(datasourceID), signedInUser, skipCache)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			ds = &models.DataSource{Name: dsName, Id: -100}
+		}
+
+		if ds == nil && dsName != "__expr__" {
+			return nil, fmt.Errorf("no datasource reference found")
+		}
+
+		if dsName == "" {
+			model["datasource"] = ds.Name
+		}
+
+		i, ok := model["datasourceId"]
+		if !ok {
+			model["datasourceId"] = ds.Id
+		} else {
+			datasourceID, ok := i.(int64)
+			if !ok || datasourceID == 0 {
+				model["datasourceId"] = ds.Id
+			}
+		}
+
+		i, ok = model["orgId"] // GEL requires orgID inside the query JSON
+		if !ok {
+			model["orgId"] = alertDefinition.OrgId
+		} else {
+			orgID, ok := i.(int64)
+			if !ok || orgID == 0 {
+				model["orgId"] = alertDefinition.OrgId
+			}
+		}
+
+		const defaultMaxDataPoints = 100
+		var maxDataPoints int64
+		i, ok = model["maxDataPoints"] // GEL requires maxDataPoints inside the query JSON
+		if !ok {
+			maxDataPoints = defaultMaxDataPoints
+		} else {
+			maxDataPoints, ok = i.(int64)
+			if !ok || maxDataPoints == 0 {
+				maxDataPoints = defaultMaxDataPoints
+			}
+		}
+		query.MaxDataPoints = maxDataPoints
+
+		// intervalMS is calculated by the frontend
+		// should we do something similar?
+		const defaultIntervalMs = 1000
+		var intervalMs int64
+		i, ok = model["intervalMs"] // GEL requires intervalMs inside the query JSON
+		if !ok {
+			intervalMs = defaultIntervalMs
+		} else {
+			intervalMs, ok = i.(int64)
+			if !ok || i == 0 {
+				intervalMs = defaultIntervalMs
+			}
+		}
+		query.Interval = time.Duration(intervalMs) * time.Millisecond
+
+		blob, err := json.Marshal(model)
+		if err != nil {
+			return nil, fmt.Errorf("unable to marshal query model %w", err)
+		}
+		rm := json.RawMessage{}
+		err = rm.UnmarshalJSON(blob)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal query model %w", err)
+		}
+		query.JSON = rm
```

Comment:
```suggestion
		if query.JSON, err = json.Marshal(model); err != nil {
			return nil, fmt.Errorf("unable to marshal query model %w", err)
		}
```

Reply:
It was too early in the morning and didn't have a lot of coffee when I wrote this!